### ISSUE TITLE
`dnPhyloCTMC` documentation

### ISSUE THREAD

**Issue by ms609** [revbayes/revbayes#486]
I'm new to RevBayes, and have been trying to get my head around how I can interact with `dnPhyloCTMC()`.  I've gleaned some aspects of its behaviour from the tutorial pages, but the entry in the [reference pages](https://revbayes.github.io/documentation/dnPhyloCTMC.html) is somewhat minimal.

- One thing I didn't see listed there was a `.clamp()` method. Perhaps `.clamp()` is conceptually an operator (I've seen it listed in that category elsewhere), but its method-like syntax meant I expected to see it on the reference page, and there might be value in stating its supported arguments there.  I couldn't find a separate reference page for `.clamp()`, so it's hard to infer its behaviour.
- Another thing that would be helpful to have documented is the 'return value'.  I suspect that I'm thinking of this in the wrong way, and the "domain type" gives the answer to the question I'm trying to ask, but I don't get from the function page or the [domain page](https://revbayes.github.io/documentation/AbstractHomologousDiscreteCharacterData.html) a clear picture of what exactly is returned, or what the pathway is to get from the distribution to the likelihood of the clamped data.

Would it be possible for the documentation to be expanded with some pointers on those topics?

**Comment by davidcerny** [comment]
Hi Martin,

Perhaps the [tutorial on graphical models](https://revbayes.github.io/tutorials/intro/graph_models) and how to specify them in Rev would be a good place to start. [This published protocol](https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/cpbi.22) might be helpful, too. In Rev, any stochastic node (i.e., a draw from a distribution) can be clamped to a fixed value of the appropriate type; draws from `dnPhyloCTMC()` are no different in this respect.

The return type of `dnPhyloCTMC()` is indeed given in the documentation under the "Domain Type". I.e., a draw from this distribution is an object of class `AbstractHomologousDiscreteCharacterData`. That also immediately answers the question of what are the expected arguments of its `.clamp()` method: a draw from this distribution can only be clamped to another `AbstractHomologousDiscreteCharacterData` object. Again, in principle this is no different from much simpler distributions such as `dnNormal()`, whose domain type is `Real`, and where a draw can accordingly only be clamped to another `Real`.

The likelihood of the clamped `AbstractHomologousDiscreteCharacterData` is computed using Felsenstein's pruning algorithm, with partial likelihoods stored for each branch of the tree. It is automatically outputted by both the `mnFile` and `mnScreen` monitors. It is true that for`dnPhyloCTMC()`, we can't compute it with the same simplicity as for simpler distributions such as `dnNormal()`:

```
x ~ dnNormal(0.0, 1.0)
x.clamp(-4.3)
x.lnProbability()
```

If, for some reason, you wish to compute the character likelihood given a fixed vector of parameter values (tree with branch lengths, exchangeabilities, stationary frequencies, site rates, etc.), you can always do it by running a dummy single-iteration MCMC analysis:

```
mymodel = model(tree)
monitors = VectorMonitors()
monitors.append( mnScreen(printgen=1) )
mymcmc = mcmc(mymodel, monitors, moves)
mymcmc.run(generations=1)
```

The value is then given in the 0th-iteration row and in the "Likelihood" column.

**Comment by ms609** [comment]
Thanks for this helpful information, which I'll integrate into a documentation pull request.

`dnPhyloCTMC()` does contain a `.lnProbability()` method; is there a reason to prefer the single-iter MCMC approach to using this?

**Comment by davidcerny** [comment]
@ms609 Never mind! I got that wrong, because I was trying to call `.lnProbability()` on a vector of draws (`AbstractHomologousDiscreteCharacterData[]`) rather than a single draw. So the whole second half of my comment above can be ignored; computing densities is just as straightforward for `dnPhyloCTMC()` as it is for simpler distributions.

**Comment by ms609** [comment]
Yes, this was closed by #487 – sorry not to catch this myself, I'd assumed the auto-closing keyword in the PR header would do the job.
